mongo/integration/collection_test.go

```diff
@@ -778,15 +779,140 @@ func TestCollection(t *testing.T) {
 		})
 		mt.Run("options", func(mt *mtest.T) {
 			initCollection(mt, mt.Coll)
-			opts := options.FindOne().SetComment("here's a query for ya")
-			res, err := mt.Coll.FindOne(mtest.Background, bson.D{{"x", 1}}, opts).DecodeBytes()
+
+			// Create an index for Hint, Max, and Min options
+			indexView := mt.Coll.Indexes()
+			indexName, err := indexView.CreateOne(context.Background(), mongo.IndexModel{
+				Keys: bson.D{{"x", 1}},
+			})
+			assert.Nil(mt, err, "CreateOne error: %v", err)
+
+			mt.ClearEvents()
+			expectedCollation := options.Collation{Locale: "en_US"}
+			expectedComment := "here's a query for ya"
+			expectedMax := bson.D{{"x", int32(5)}}
+			expectedMin := bson.D{{"x", int32(0)}}
+			expectedProjection := bson.D{{"x", int32(1)}}
+			expectedSort := bson.D{{"x", int32(1)}}
+			// SetCursorType is excluded because tailable cursors can't be used with limit -1,
```

Comment:
I filed https://jira.mongodb.org/browse/GODRIVER-1775 last week to address options like `CursorType` and `MaxAwaitTime`. They don't make any sense for a FindOne operation, so we should deprecate/remove them. Might be useful to work the ticket number into the comments here so future readers are aware of that.

Reply:
done